### DISCUSSION TITLE
ElementPropertyReadFile is buggy

### DISCUSSION THREAD

**Discussion by RWTHLHK** [idaholab/moose#28974] [Q&A Modules: Solid mechanics]
Now I am trying to perform crystal plasticity simulations using polycrystals. But I always get a weird error, as shown in the figure, there are exactly 271 grains(I check this in peacock, there are indeed 271 blocks named as grain-i, i is from 1 to 271). However, it always reports this error:
`Element block id 271 greater than than total number of blocks in mesh: 271. Blocks should be numbered consecutively.
![Screenshot from 2024-10-31 02-09-34](https://github.com/user-attachments/assets/ce513ee6-cce7-4b6d-97fc-f9e5c6d7576d)
`
And there are indeed 271 lines in EulerAngles.txt:
[UserObjects]
  [./prop_read]
    type = ElementPropertyReadFile
    prop_file_name = 'EulerAngles.txt'
    # Enter file data as prop#1, prop#2, .., prop#nprop
    nprop = 3
    read_type = block
    nblock= 271
  [../]
[]

**Comment by RWTHLHK** [comment]
![Screenshot from 2024-10-31 02-09-34](https://github.com/user-attachments/assets/6743656d-738b-4c97-ad41-7c73e68eb104)
Sry I didnt upload the figure successfully. This is the polycrystal I used. The maximum block id is grain-271.

**Reply by GiudGiud** [reply]
add a dummy line at the beginning of the prop_file_names file (EulerAngle.txt)

**Reply by RWTHLHK** [reply]
Thanks for reply! I found the reason is in my .e mesh file, block id counts from 1, but in moose it should count from 0. I just fixed it.

**Reply by GiudGiud** [reply]
I ll add a note in the error message. Thanks for letting us know
